media/magicavoxel_vox.ksy

```diff
@@ -0,0 +1,187 @@
+meta:
+  id: vox
+  file-extension: vox
+  application: MagicaVoxel
+  endian: le
+  license: MIT
+  title: MagicaVoxel File
+doc-ref: 'https://ephtracy.github.io/ MagicaVoxel Homepage'
+doc-ref: 'https://github.com/ephtracy/voxel-model/blob/master/MagicaVoxel-file-format-vox.txt Format Description'
+seq:
+  - id: magic
+    contents: 'VOX '
+  - id: version
+    type: u4
+    doc: 150 expected
+  - id: main
+    type: chunk
+types:
+  chunk:
+    seq:
+      - id: chunk_id
+        type: u4be
+        enum: chunk_type
+      - id: num_bytes_of_chunk_content
+        type: u4
+      - id: num_bytes_of_children_chunks
+        type: u4
+      - id: chunk_content
+        if: num_bytes_of_chunk_content != 0
+        size: num_bytes_of_chunk_content
+        type:
+          switch-on: chunk_id
+          cases:
+            'chunk_type::pack': pack
+            'chunk_type::size': size
+            'chunk_type::xyzi': xyzi
+            'chunk_type::rgba': rgba
+            'chunk_type::matt': matt
+      - id: children_chunks
+        if: num_bytes_of_children_chunks != 0
+        type: chunk
+        repeat: eos
+  pack:
+    seq:
+      - id: num_models
+        type: u4
+  size:
+    seq:
+      - id: size_x
+        type: u4
+      - id: size_y
+        type: u4
+      - id: size_z
+        type: u4
+  xyzi:
+    seq:
+      - id: num_voxels
+        type: u4
+      - id: voxels
+        type: voxel
+        repeat: expr
+        repeat-expr: num_voxels
+  rgba:
+    seq:
+      - id: colors
+        type: color
+        repeat: expr
+        repeat-expr: 256
+  voxel:
+    seq: 
+      - id: x
+        type: u1
+      - id: y
+        type: u1
+      - id: z
+        type: u1
+      - id: color_index
+        type: u1
+  color:
+    seq:
+      - id: r
+        type: u1
+      - id: g
+        type: u1
+      - id: b
+        type: u1
+      - id: a
+        type: u1
+  matt:
+    seq:
+    - id: id
+      type: u4
+    - id: material_type
+      type: u4
+      enum: material_type
+    - id: material_weight
+      type: f4
+    - id: property_bits
+      type: u4
+      # .to_i not implemented for C# runtime yet
+      #enum: property_bits_type
```

Comment:
It's better to use separate `type` that would look like that for bitmask:

```yaml
types:
  property_bits_type:
    seq:
      - id: reserved1
        type: b1
      - id: has_is_total_power
        type: b1
      - id: has_glow
        type: b1
      - id: has_power
        type: b1
      # ...
```

instead of enums or manual extraction of flags using the value instances.